diff -Nur PK2CMD_SourceV1_23_00/pk2cmd/pk2cmd/cmd_app.cpp PK2CMD_SourceV1_23_00.new/pk2cmd/pk2cmd/cmd_app.cpp
--- PK2CMD_SourceV1_23_00/pk2cmd/pk2cmd/cmd_app.cpp	2022-06-15 17:36:26.000000000 +0800
+++ PK2CMD_SourceV1_23_00.new/pk2cmd/pk2cmd/cmd_app.cpp	2022-06-19 04:34:56.390265315 +0800
@@ -86,12 +86,8 @@
 		return;
 	}
 	else
-	{ // no -B, check PATH
-		_tsearchenv_s("PK2DeviceFile.dat", "PATH", tempString);
-		if (_tcslen(tempString) < 17)
-		{
-			_tcsncpy_s(tempString, "PK2DeviceFile.dat", 17);
-		}
+	{ // no -B
+		_tcsncpy_s(tempString, "/usr/share/pk2/PK2DeviceFile.dat", 32);
 	}
 	if (!PicFuncs.ReadDeviceFile(tempString))
 	{
@@ -504,10 +500,18 @@
 			ret = Pk2BootLoadFuncs.ReadHexAndDownload(tempString, &PicFuncs, pk2UnitIndex);
 			if (!ret)
 			{
-				printf("Error opening hex file.\n");
-				fflush(stdout);
-				ReturnCode = OPFAILURE;
-				return true; // download command found
+				_TCHAR temp2String[MAX_PATH] = "";
+				_tcsncpy_s(temp2String, tempString, MAX_PATH-1);
+				_tcsncpy_s(tempString, "/usr/share/pk2/", 15);
+				_tcscat_s(tempString, temp2String);
+				ret = Pk2BootLoadFuncs.ReadHexAndDownload(tempString, &PicFuncs, pk2UnitIndex);
+				if (!ret)
+				{
+					printf("Error opening hex file.\n");
+					fflush(stdout);
+					ReturnCode = OPFAILURE;
+					return true; // download command found
+				}
 			}
 			ret = Pk2BootLoadFuncs.ReadHexAndVerify(tempString, &PicFuncs);
 			if (!ret)
